content/articles/Developing-simple-Blockchain-app-In-flutter/index.md

```diff
@@ -0,0 +1,216 @@
+```dart
+class Blockchain {
+    MqttClient _mqttClient = new MqttClient();
+
+AsyncTask<Void, Void, String>  
+ asyncTask = new AsyncTask<Void, Void, String>() {
+
+ @Override  protected String doInBackground(Void... params) {
+
+return null;        }
+
+};
+```
+Now we can run the MqttClient instance on a different thread. However, if we don't register it with something that can handle the messages, it will never receive any messages. Let's add an instance of the MqttMessageHandler class to our application and register our _mqttClient instance with it.
+```dart
+class Blockchain {
+
+    MqttClient _mqttClient = new MqttClient();
+
+AsyncTask<Void, Void, String> asyncTask = new AsyncTask<Void, Void, String>() {
+
+          @Override  protected String doInBackground(Void... params) {
+
+return null;         }
+
+};
+
+MqttMessageHandler messageHandler = new MqttMessageHandler();
+
+_mqttClient.setHandler(messageHandler);
+```
+Now our _mqttClient instance will receive messages from the blockchain network. Finally, let's add some code to our save() function to save the blockchain data to a file. 
+```dart
+class Blockchain {
+
+    MqttClient _mqttClient = new MqttClient();
+
+AsyncTask<Void, Void, String> asyncTask = new AsyncTask<Void, Void, String>() {
+
+ @Override  protected String doInBackground(Void... params) {
+return null;         }
+@Override  protected void onPostExecute(String result) {
+getBlockHeight();
+Blockchain network = new Blockchain();
+   _mqttClient.connect("myDevices");
+```
+Now we can build our blockchain network by adding devices to it one at a time after they have registered with the same MQTT broker. Fortunately, flutter provides us with an easy way to do this. All we have to do is add a button and call the addDevice() function when it is clicked.
+```dart
+void main() {
+runApp(new MyApp());
+
+}
+class MyApp extends StatelessWidget {
+Widget build(BuildContext context) {
+return new MaterialApp(
+ home: new Scaffold(
+ appBar: new AppBar(
+ title: new Text("My Blockchain"),
+ ),
+ body: new HomePage(), );
+       Button(
+          onClick: () {
+           network.addDevice(); 
+});
+} } 
+```
+Notice that we are connecting to my Devices, which is the name of my MQTT broker. You will need to change this to match the name of your broker. Now when you run the application on different devices, they will all be connected to the same blockchain network and will share the same blockchain data.
+### Conclusion
+Congratulations! You have now created a flutter application that uses a dart blockchain. In Part 2 of this series, we will learn how to create a proof of work algorithm and add it to our blockchain so that other devices will be able to join the network.
```

Comment:
At Section we generally do not encourage series articles. However, you can suggest a more advanced application of the same topic as a new article and refer readers to this article as a prerequisite to following along.